core/services/pg/q.go

```diff
@@ -141,6 +136,14 @@ func (q Q) WithOpts(qopts ...QOpt) Q {
 }
 
 func (q Q) Context() (context.Context, context.CancelFunc) {
+	if q.QueryTimeout > 0 {
+		ctx := context.Background()
+		if q.ParentCtx != nil {
+			ctx = q.ParentCtx
+		}
```

Comment:
nit/ Could flip these to make a little shorter :shrug: 
```suggestion
		ctx := q.ParentCtx
		if ctx == nil {
			ctx = context.Background()
		}
```